In [13]:
import sqlite3
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import json


con = sqlite3.connect("db.sqlite3")
df = pd.read_sql_query("select DATE,RF from weather_data",con)
con.close()
df = df.copy()
df = df.loc[:,['date','RF']]
df['Date'] = pd.to_datetime(df['date'], errors='coerce')
df = df.loc[:,['Date','RF']]
df.index = pd.to_datetime(df.index,unit='D')
df = df.asfreq('W').ffill()
df['P'] = df['RF'].apply(lambda x: 1 if x >= 1 else 0)
df = df.loc[:,['Date','P']]
df = df.asfreq('W').ffill()
df.reset_index(drop=True, inplace=True)
df = df.sort_values('Date')
df.reset_index(drop=True, inplace=True)
l = ['year','weekofyear']
df = df.join(pd.concat((getattr(df['Date'].dt, i).rename(i) for i in l), axis=1))
df = df.loc[:,['weekofyear','P']]
df['sec'] = df.P.shift(1)
def analysis(df):
    if (df['P'] == 0 and df['sec']==0):
        return 'DD'
    elif (df['P'] == 0 and df['sec']==1):
        return 'DW'
    elif (df['P'] == 1 and df['sec']==0):
        return 'WD'
    elif (df['P'] == 1 and df['sec']==1):
        return 'WW'
df['N'] = df.apply(analysis,axis = 1)
#df = pd.get_dummies(df.set_index('NO')['P']).max(level=0).reset_index()
df = df.groupby('weekofyear').N.value_counts().unstack().fillna(0)
df['N0'] = df['DD'] + df['DW']
df['N1'] = df['WD'] + df['WW']
df['PDD'] = df['DD'] / df['N0']
df['PDW'] = df['DW'] / df['N0']
df['PWD'] = df['WD'] / df['N1']
df['PWW'] = df['WW'] / df['N1']
df['PD'] = df['DD'] + df['DW'] / df['N0'] + df['N1']
df['PW'] = df['WD'] + df['WW'] / df['N0'] + df['N1']
df = df.round(2)
print(df)







N             DD    DW   WD   WW    N0    N1   PDD   PDW   PWD   PWW     PD  \
weekofyear                                                                    
1           15.0   1.0  1.0  0.0  16.0   1.0  0.94  0.06  1.00  0.00  16.06   
2           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
3           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
4           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
5           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
6           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
7           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
8           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
9           17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
10          17.0   0.0  0.0  1.0  17.0   1.0  1.00  0.00  0.00  1.00  18.00   
11          17.0   0.0  0.0  1.0  17.0   1.0  1.00  

#    test

In [114]:
list_of_dates = ['2012-12-31', '2012-12-29', '2012-12-30']
df = pd.DataFrame({'ArrivalDate': pd.to_datetime(list_of_dates)})

L = ['year', 'month', 'day', 'dayofweek', 'dayofyear', 'weekofyear', 'quarter']
df = df.join(pd.concat((getattr(df['ArrivalDate'].dt, i).rename(i) for i in L), axis=1))

print(df)


  ArrivalDate  year  month  day  dayofweek  dayofyear  weekofyear  quarter
0  2012-12-31  2012     12   31          0        366           1        4
1  2012-12-29  2012     12   29          5        364          52        4
2  2012-12-30  2012     12   30          6        365          52        4


# ###p1 = df_p[0:52].tolist()
p2 = df_p[52:104].tolist()
p3 = df_p[104:156].tolist()
p4 = df_p[156:208].tolist()
p5 = df_p[208:260].tolist()
p6 = df_p[260:312].tolist()
p7 = df_p[312:364].tolist()
p8 = df_p[364:416].tolist()
p9 = df_p[416:468].tolist()
p10 = df_p[468:520].tolist()
p11 = df_p[520:572].tolist()
p12 = df_p[572:624].tolist()
p13 = df_p[624:676].tolist()
p14 = df_p[676:728].tolist()
p15 = df_p[728:780].tolist()
p16 = df_p[780:832].tolist()
p17 = df_p[832:884].tolist()
p18 = df_p[884:936].tolist()

  ArrivalDate  year  month  day  dayofweek  dayofyear  weekofyear  quarter
0  2012-12-31  2012     12   31          0        366           1        4
1  2012-12-29  2012     12   29          5        364          52        4
2  2012-12-30  2012     12   30          6        365          52        4

In [7]:
import sqlite3
import pandas as pd
import numpy as np
import random as rm
import xlrd

con = sqlite3.connect("db.sqlite3")
df = pd.read_sql_query("select DATE,RF from weather_data",con)
con.close()
df = df.copy()
df = df.loc[:,['date','RF']]
print(df)

                     date   RF
0     1993-01-01 00:00:00  0.0
1     1993-01-02 00:00:00  0.0
2     1993-01-03 00:00:00  0.0
3     1993-01-04 00:00:00  0.0
4     1993-01-05 00:00:00  0.0
5     1993-01-06 00:00:00  0.0
6     1993-01-07 00:00:00  0.0
7     1993-01-08 00:00:00  0.0
8     1993-01-09 00:00:00  0.0
9     1993-01-10 00:00:00  0.0
10    1993-01-11 00:00:00  0.0
11    1993-01-12 00:00:00  0.0
12    1993-01-13 00:00:00  0.0
13    1993-01-14 00:00:00  0.0
14    1993-01-15 00:00:00  0.0
15    1993-01-16 00:00:00  0.0
16    1993-01-17 00:00:00  0.0
17    1993-01-18 00:00:00  0.0
18    1993-01-19 00:00:00  0.0
19    1993-01-20 00:00:00  0.0
20    1993-01-21 00:00:00  0.0
21    1993-01-22 00:00:00  0.0
22    1993-01-23 00:00:00  0.0
23    1993-01-24 00:00:00  0.0
24    1993-01-25 00:00:00  0.0
25    1993-01-26 00:00:00  0.0
26    1993-01-27 00:00:00  0.0
27    1993-01-28 00:00:00  0.0
28    1993-01-29 00:00:00  0.0
29    1993-01-30 00:00:00  0.0
...                   ...  ...
6544  20